components/workouts/ExerciseList.jsx

```diff
@@ -0,0 +1,154 @@
+   top: 14px;
+   right: 20px;
+`;
+
+const Buttontext = styled.Text`
+  font-family: 'Montserrat_500Medium';
+  font-size: 18px;
+`;
+
+const parseItems = (items) => {
+  // Sort names alphabetically
+  items.sort((a, b) => a.name.localeCompare(b.name));
+
+  // Group by first letter of each name
+  const bucketData = items.reduce((accumulator, item) => {
+    const bucket = item.name[0].toUpperCase();
+
+    // If this is the first time we've seen this letter, create a bucket
+    if (!accumulator[bucket]) {
+      accumulator[bucket] = [item];
+    } else {
+      accumulator[bucket].push(item);
+    }
+
+    return accumulator;
+  }, {});
+  return bucketData;
+};
+
+const renderHeader = ({ section }) => (
+  <SectionHeader>{section.title}</SectionHeader>
+);
+
+const ExerciseList = (props) => {
+  const { items } = props;
+
+  const parsedItems = parseItems(items);
+  const [search, setSearch] = React.useState('');
+  const [filteredDataSource, setFilteredDataSource] = React.useState([]);
+  const [masterDataSource, setMasterDataSource] = React.useState([]);
+  React.useEffect(() => {
+    setFilteredDataSource(parsedItems);
+    setMasterDataSource(parsedItems);
+  }, []);
+  const searchFilterFunction = (text) => {
+    // Check if searched text is not blank
+    if (text) {
+      // Inserted text is not blank
+      // Filter the masterDataSource
+      // Update FilteredDataSource
+      const firstLetter = text[0];
+      if (masterDataSource[firstLetter]) {
+        const newData = masterDataSource[firstLetter].filter((item) => {
+          const itemData = item.name
+            ? item.name.toUpperCase()
+            : ''.toUpperCase();
+          const textData = text.toUpperCase();
+          return itemData.indexOf(textData) > -1;
+        });
+        const finished = {};
+        finished[firstLetter] = newData;
+        setFilteredDataSource(finished);
+        setSearch(text);
+      } else {
+        setFilteredDataSource(masterDataSource);
+        setSearch(text);
+      }
+    } else {
+      // Inserted text is blank
+      // Update FilteredDataSource with masterDataSource
+      setFilteredDataSource(masterDataSource);
+      setSearch(text);
+    }
+  };
+  const [exerciseCount, setExerciseCount] = React.useState(0);
+  const [addedExercises] = React.useState([]);
+  const renderCard = ({ item, index }) => (
+    <ExerciseItem
+      name={item.name}
+      subtext={item.subtext}
+      selected={item.selected}
+      onPress={() => {
+        const temp = { ...masterDataSource };
+        const { selected } = temp[item.name[0]][index];
+        temp[item.name[0]][index].selected = !selected;
+        setMasterDataSource(temp);
+        (temp[item.name[0]][index].selected === true) ? setExerciseCount(exerciseCount + 1)
+          : setExerciseCount(exerciseCount - 1);
+        (temp[item.name[0]][index].selected === true) ? addedExercises.push(item)
+          : addedExercises.splice(addedExercises.indexOf(item), 1);
```

Comment:
Is there a reason you chose to constantly update this rather than calculate it at the end?

Reply:
I just thought it would be easy to do it this way without using too much rendering power (not sure though). If there is a need later as we are tidying up code for it to be a function, then I can do that.

Reply:
Yeah this is fine for now, we can go back and clean it up after our presentation